### ISSUE TITLE
2FA not working for admin account

### ISSUE THREAD

**Issue by dartmann** [nextcloud/twofactor_gateway#149]
**Nextcloud version**: 14.0.4.2
**Operating system and version**: Ubuntu 18.04.1 LTS
**Apache version**: Apache/2.4.29
**PHP version**: 7.2.10

**The issue you are facing**: 2FA is not working for my Nextcloud admin account. 2FA for normal users works as expected. When enabling the feature it does not show any error but after logging out my admin account and logging back in, I am not asked for a TOTP code.
I already tried to solve the issue by removing the 2FA app, but without success.

**Is this the first time you’ve seen this error?**: I can not see any error message.

**Steps to replicate it**: I can not provide steps to replicate, because I think its a local problem (many out there are surely using 2FA for their admin account).

**The output of your Nextcloud log in Admin > Logging**: nothing provided there.

**The output of your config.php file**:

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "keller2",
            "192.168.178.64",
            "keller2.8pzkpdxsetbyyr3f.myfritz.net",
            "keller2.fritz.box",
            "h6h93vrweb7nz2vb.myfritz.net"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/keller2\/nextcloud",
        "dbtype": "mysql",
        "version": "14.0.4.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "theme": "",
        "loglevel": 2,
        "maintenance": false,
        "trashbin_retention_obligation": "auto",
        "updatechecker": false,
        "integrity.check.disabled": false,
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "tls",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

**The output of your Apache/system log in /var/log/____**: nothing to post

**Comment by dartmann** [comment]
Hi @ChristophWurst 
thanks for your reply.

I have access to the server's shell and the response of the command is as follows:
```
Two-factor authentication is enabled for user admin

Enabled providers:
- backup_codes
- totp
```

**Comment by ChristophWurst** [comment]
This is quite unexpected. What version of this app do you use?

**Comment by dartmann** [comment]
The latest stable: 1.5.0

**Comment by dartmann** [comment]
I just set up a new nextcloud instance and, of course, it is working without problems on all accounts, admins and normal users.
To be honest, my setup is pretty old: originally I used owncloud and migrated to nextcloud some major releases ago. So a possible cause for this behavior could be because of my often updated and migrated system. Nevertheless, there is no hard evidence for this and all other things are working very fine.
Is there any other thing I could take a look at? Some logs or testing?
Thank you for your help @ChristophWurst 👍 I really appreciate it.

**Comment by ChristophWurst** [comment]
Yeah, still, I also have an instance that is ages old but everything works as smooth as a new instance :wink:

Do you see the TOTP settings when you go to your personal security settings?

**Comment by dartmann** [comment]
Yes, both a normal user as well as the admin see the TOTP settings in their personal security settings view as follows (German):
<img width="434" alt="grafik" src="https://user-images.githubusercontent.com/4301606/49302145-86f97f00-f4c6-11e8-89bd-d708d8d83f12.png">

**Comment by ChristophWurst** [comment]
Wait a minute … this repo is for the `twofactor_gateway` app. What app are you having issues with? https://github.com/nextcloud/twofactor_totp?

**Comment by dartmann** [comment]
@ChristophWurst sorry, I overlooked this one... Yes, I have issues with the TOTP app my referenced. Do I have to open a new Issue in the corresponding repo, or are you able to move this one? Sorry again 😞
